WalletWasabi.Fluent/Controls/MultiSelector.axaml.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using Avalonia;
+using Avalonia.Controls;
+using DynamicData;
+using ReactiveUI;
+
+namespace WalletWasabi.Fluent.Controls;
+
+public class MultiSelector : ItemsControl, IDisposable
+{
+    private readonly CompositeDisposable _disposables = new();
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Using DisposeWith")]
+	public MultiSelector()
+    {
+        var changes = this
+            .WhenAnyValue(x => x.Items)
+            .WhereNotNull()
+            .Select(x => x.Cast<ISelectable>())
+            .Select(x => x.AsObservableChangeSet())
+            .Switch();
+        
+        var isSelected = changes
+            .AutoRefresh(x => x.IsSelected)
+            .ToCollection()
+            .Select(GetSelectionState);
+
+        var isSelectedSubject = new BehaviorSubject<bool?>(false);
+        isSelected.Subscribe(isSelectedSubject);
+        
+        Toggle = ReactiveCommand.CreateFromObservable(() =>
+            {
+                var currentValue = isSelectedSubject.Value;
+                return ToggleChildrenSelection(changes, GetNextValue(currentValue));
+            })
+            .DisposeWith(_disposables);
+
+        IsChecked = isSelected
+            .CombineLatest(Toggle.IsExecuting)
+            .Where(x => !x.Second)
+            .Select(x => x.First)
+            .Replay(1)
+            .RefCount();
+    }
+
+    public ReactiveCommand<Unit, IReadOnlyCollection<ISelectable>> Toggle { get; }
+
+    public IObservable<bool?> IsChecked { get; }
```

Comment:
Since this CheckBox is three-state, null is used for the "ndetermined" value. That happens when neither all or none of items are selected. Eg: 
![image](https://user-images.githubusercontent.com/3109851/193066393-8161242a-2c45-4cbb-b332-a26ad5218314.png)